src/main/java/io/mosip/iiitb/odte/utils/RSACryptoTool.java

```diff
@@ -0,0 +1,110 @@
+package io.mosip.iiitb.odte.utils;
+
+import com.google.inject.Inject;
+import io.mosip.iiitb.odte.config.OnDemandAppConfig;
+import org.slf4j.Logger;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.OAEPParameterSpec;
+import javax.crypto.spec.PSource;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.*;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.MGF1ParameterSpec;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+public class RSACryptoTool {
+    final PrivateKey pk;
+    Cipher cipher = null;
+    final Logger logger;
+
+    @Inject
+    public RSACryptoTool(
+        OnDemandAppConfig onDemandAppConfig,
+        Logger logger
+    ) throws NoSuchAlgorithmException, InvalidKeySpecException, IOException {
+        this.logger = logger;
+        String privateKeyFileLocation = onDemandAppConfig.privateKeyFileLocation();
```

Comment:
Don't keep the private key file in plain text, atlest lets keep it in a p12 file which is password protected.